lib/mongoid_rails_migrations/active_record_ext/migrations.rb

```diff
@@ -352,6 +395,7 @@ def status
       puts "-" * 50
       up_migrations = migrated.to_set
       migrations.each do |migration|
+        next unless client_match?(migration)
```

Comment:
Yep, that's true, now it loads the migration file to detect the client and for migrations loading files that does not exist anymore it will fail.

My POV on this is that we should not keep hundreds migration files on a project like in this example:
```
$ RAILS_ENV=staging bundle exec rails db:migrate:status | grep up | wc -l
644
```
I try to keep only the last 3 years of migrations files on the same project, it works well and still contains a lot of migration files:
```
$ rails db:migrate:status | grep up | wc -l
157
```
Moreover, keeping legacy code in codebase is misleading IMO and can lead newcomers to use deprecated methods in their own migrations.

> It could maybe be handled by grouping migrations for the same non-default client in a subfolder? This way it'd be organized and we could infer the client thanks to the folder name without loading the migration?

That's a good idea but I fear that that could be a bit confusing for users in the end as we've already [multi folders support ](https://github.com/adacosta/mongoid_rails_migrations/blob/master/lib/mongoid_rails_migrations/active_record_ext/migrations.rb#L362) that is not only doing more than merging all migrations in the same array.

I'm not saying no, just explaining why I choose this opinionated way ;)

WDYT @jarthod ?

Reply:
Both good points, quick question: does it fail on our project? because it's true we now need to load the migration class but we don't need to run the `up` or `down` which are the ones containing the legacy code usually. So unless there's some deprecated code used in the migration class directly (like to build a constant), this has low change to fail.

Anyway we can discuss this again after we address the bigger concern of my review.

Reply:
Yes, it fails for old migrations requiring files that are not present in the project anymore. As I explained above, it does not fail for 3 years old migrations but for 10 years old yes.

Reply:
Ok thanks, for our project I don't care much and I agree we could just remove the old or failing migrations, but what I don't like is that this is gonna fail for anybody using the gem, even if they don't use sharding (nobody will). So I would say it's ok to keep it like this IF you change the code to only load the migration when using MONGOID_CLIENT_NAME, does that sound possible? so for 99.9% of users this should not change anything.

Reply:
That's a good idea but seems not feasible as when we do not use `MONGOID_CLIENT_NAME`, we need to know if the migration targets the default client or not, if not we've to skip the migration. If you think that's a pain, we can go for the folder way.

We could also require to enable a `sharding` option in config in order to make the client check only when the `sharding` is turned on.

Reply:
Mmm indeed, but maybe we can totally ignore this when MONGOID_CLIENT_NAME is not passed? Which means force people to always use "MONGOID_CLIENT_NAME" when using sharding? (maybe a bit dangerous). Otherwise I guess the folders seems best indeed.